qemu/tests/block_hotplug.py

```diff
@@ -185,16 +218,22 @@ def get_new_disk(disk1, disk2):
                                   logging.info)
             utils_test.run_virt_sub_test(test, params, env, sub_type)
 
-        for num in xrange(blk_num):
-            error_context.context("Unplug block device (iteration %d)" % iteration,
-                                  logging.info)
-            disks_before_unplug = find_disk(vm, get_disk_cmd)
-            device_list[num].unplug(vm.monitor)
-            device_list[num].verify_unplug("", vm.monitor)
-            unplug_status = utils_misc.wait_for(lambda: len(get_new_disk(find_disk(vm, get_disk_cmd),
-                                                disks_before_unplug)) != 0, pause)
-            if not unplug_status:
-                test.fail("Failed to unplug disks")
+        error_context.context("Unplug block device (iteration %d)" % iteration,
+                              logging.info)
+        if controller_list:
+            for controller in controller_list:
+                controller_id = controller.get_param("id")
+                if controller_id in controller_device_dict.keys():
+                    for device in controller_device_dict[controller_id]:
+                        unplug_status = unplug_device(vm, get_disk_cmd, device)
+                        if not unplug_status:
+                            test.fail("Failed to unplug disks")
+                controller.unplug(vm.monitor)
+        else:
```

Comment:
Some device would be missed to be unplugged with the current logic.

Reply:
@luckyh ,have updated, please help to review again.